pkg/service/update/update.go

```diff
@@ -0,0 +1,145 @@
+package updateBinary
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+
+	"github.com/charmbracelet/glamour"
+	"go.keploy.io/server/utils"
+	"go.uber.org/zap"
+)
+
+// GitHubRelease holds information about the GitHub release.
+type GitHubRelease struct {
+	TagName string `json:"tag_name"`
+	Body    string `json:"body"`
+}
+
+// updater manages the updating process of the Keploy binary.
+type updater struct {
+	logger *zap.Logger
+}
+
+// NewUpdater initializes a new updater instance.
+func NewUpdater(logger *zap.Logger) Updater {
+	return &updater{
+		logger: logger,
+	}
+}
+
+// Updater defines the contract for updating the Keploy binary.
+type Updater interface {
+	UpdateBinary()
+}
+
+var ErrGitHubAPIUnresponsive = errors.New("GitHub API is unresponsive")
+
+// UpdateBinary initiates the update process for the Keploy binary file.
+func (u *updater) UpdateBinary() {
+	currentVersion := utils.Version
+
+	isDockerCmd := len(os.Getenv("IS_DOCKER_CMD")) > 0
+	if isDockerCmd {
+		u.logger.Info("Please Pull the latest Docker image of Keploy")
+		return
+	}
+
+	releaseInfo, err := utils.GetLatestGitHubRelease()
+	latestVersion := releaseInfo.TagName
+	changelog := releaseInfo.Body
+
+	if err != nil {
+		if err == ErrGitHubAPIUnresponsive {
+			u.logger.Error("GitHub API is unresponsive. Update process cannot continue.")
+		} else {
+			u.logger.Error("Failed to fetch latest GitHub release version", zap.Error(err))
+		}
+		return
+	}
+
+	if currentVersion == latestVersion {
+		u.logger.Info("No updates available. Current Version " + currentVersion + " " + latestVersion + " is the latest.")
+		return
+	}
+
+	u.logger.Info("Updating to Version: " + latestVersion)
+
+	arch := runtime.GOARCH
+	downloadUrl := ""
+	if arch == "amd64" {
+		downloadUrl = "https://github.com/keploy/keploy/releases/latest/download/keploy_linux_amd64.tar.gz"
+	} else {
+		downloadUrl = "https://github.com/keploy/keploy/releases/latest/download/keploy_linux_arm64.tar.gz"
+	}
+	err = u.downloadAndUpdate(downloadUrl)
+	if err != nil {
+		u.logger.Error("update failed", zap.Error(err))
+		return
+	}
+
+	u.logger.Info("Update successful ")
+
```

Comment:
Please provide a screenshot for this successful update after all the chnages

Reply:
![Uploading image.png…]()

Reply:
Kindly check